### PR TITLE
[IO-1800] Set priority

### PR DESCRIPTION
**Pull request by owencjones**
# Problem
No current access to setting priority of items

# Solution
Implemented priority setter.

# Changelog
Priority setting method in core
Tests

### CODE REVIEW

File: darwin/future/core/items/set_priority.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+from darwin.future.core.client import ClientCore
+from darwin.future.core.types.common import JSONType
+from darwin.future.data_objects.typing import UnknownType
+
+
+def set_item_priority(
+    api_client: ClientCore,
+    team_slug: str,
+    priority: int,
+    filters: Optional[Dict[str, UnknownType]] = None,
```
Comment (Nathanjp91) [inline darwin/future/core/items/set_priority.py:12]: This is not blocking at all but interested here on your thoughts on 
`filters: Optional[...] = None` vs `filters: ... = {}` Is it more pythonic to assume None's or empty objects? The former does tend to lead to requiring more typecheck guards to satisfy mypy/pylance with like `assert filters is not None` etc, which does feel more verbose than required sometimes.

File: darwin/future/tests/core/items/test_set_priority.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import pytest
+
+from darwin.exceptions import DarwinException
+from darwin.future.core.client import ClientCore
+from darwin.future.core.items.set_priority import set_item_priority
+
+
+def test_set_item_priority() -> None:
+    # Create a mock API client
+    api_client = Mock(spec=ClientCore)
```
Comment (Nathanjp91) [inline darwin/future/tests/core/items/test_set_priority.py:12]: We do have the response library for this, although not at issue by any means.

**Review by Nathanjp91 — APPROVED** [review]
Nothing blocking

File: darwin/future/core/items/set_priority.py (hunk 1 above)
Comment (Nathanjp91) [reply darwin/future/core/items/set_priority.py:12]: but obviously, it is a bit more clear that filters isn't required via the `Optional[...] = None` approach

Comment (owencjones) [reply darwin/future/core/items/set_priority.py:12]: Nah, good point, I like the default in this backend code, will take this suggestion

File: darwin/future/tests/core/items/test_set_priority.py (hunk 2 above)
Comment (owencjones) [reply darwin/future/tests/core/items/test_set_priority.py:12]: Yeah, actually I'll rewrite with that in order to maintain our test style

**Pull request merged**